src/mlpack/methods/reinforcement_learning/replay/episodic_replay.hpp

```diff
@@ -0,0 +1,271 @@
+  * @param sampledRewards Sampled rewards.
+  * @param sampledNextStates Sampled encoded next states.
+  * @param isTerminal Indicate whether corresponding next state is terminal
+  *        state.
+  */
+
+  void Random_Episode(arma::mat& episodeStates,
+                      arma::icolvec& episodeActions,
+                      arma::colvec& episodeRewards,
+                      arma::mat& episodeNextStates,
+                      arma::icolvec& isTerminal)
+  {
+    size_t upperBound = full ? capacity : position;
+    int lo = 0;
+    int high = upperBound;
+    if(!upperBound)
+    {
+      high = 1;
+    }
+    int episodeNum = math::RandInt(lo,high);
+    std::vector<arma::colvec> temp = states[episodeNum];
+    int i = 0;
+    for(auto state : temp)
+    {
+      if(i==0)
+      {
+        episodeStates = state;
+        i++;
+      }else
+      {
+        episodeStates = arma::join_rows(episodeStates, state);
+      }
+    }
+    episodeActions = arma::conv_to<arma::icolvec>::from(actions[episodeNum]);
+    episodeRewards = arma::conv_to<arma::colvec>::from(rewards[episodeNum]);
+    temp = next_states[episodeNum];
+    i = 0;
+    for(auto state : temp)
+    {
+      if(i==0)
+      {
+        episodeNextStates = state;
+        i++;
+      }else
+      {
+        episodeNextStates = arma::join_rows(episodeNextStates, state);
+      }
+    }
+    isTerminal = arma::conv_to<arma::icolvec>::from(this->isTerminal[episodeNum]);
+    }
+
+  /**
+  * Get the number of episodes in the memory.
+  *
+  * @return Actual used memory size
+  */
+
+  const size_t Size()
+  {
+    if(states[position].size()==0){
+      return full ? capacity : position;  
+    }else{
+      return full ? capacity : (position+1);  
+    }
+    return full ? capacity : position;
+  }
+
+
+  /**
+  * Get the most recently added episode.
+  * @param sampledStates Sampled encoded states.
+  * @param sampledActions Sampled actions.
+  * @param sampledRewards Sampled rewards.
+  * @param sampledNextStates Sampled encoded next states.
+  * @param isTerminal Indicate whether corresponding next state is terminal
+  *        state.
+  */
+  void Sample(arma::mat& episodeStates,
+                      arma::icolvec& episodeActions,
+                      arma::colvec& episodeRewards,
+                      arma::mat& episodeNextStates,
+                      arma::icolvec& isTerminal)
+  {
+    int episodeNum=0;
+    if(states[position].size()==0 || clear==true)
+    {
+      if(position==0)
+        episodeNum = capacity-1;
+      else
+        episodeNum = position-1;
+    }
+    else
+      episodeNum = position;
+    std::vector<arma::colvec> temp = states[episodeNum];
+    int i = 0;
+    for(auto state : temp)
+    {
+      if(i==0)
+      {
+        episodeStates = state;
+        i++;
+      }else
+      {
+        episodeStates = arma::join_rows(episodeStates, state);
+      }
+    }
+    episodeActions = arma::conv_to<arma::icolvec>::from(actions[episodeNum]);
```

Comment:
Looks like we could write an extra function for this and use it for the `Sample` and `Random_Episode`, to avoid code duplication.

Reply:
Please address this comment as well.

Reply:
Yes, So I have removed the function Random_Episode entirely and have added another argument named random. If it is true Random Sampling will happen